### PR TITLE
Drop InfluxDB, Add Tornado 5, Add Python 3.7

### PR DESCRIPTION
**Pull request by andrewrabert**
- InfluxDB was dropped in favor of [sprockets-influxdb](https://github.com/sprockets/sprockets-influxdb)
- Require Tornado>=5 and Python>=3.7
- Misc cleanup
- Require `setup.py check` pass in CI to ensure required package metadata is present.
- Require `setup.py build_sphinx` in CI to ensure docs build correctly.
- Require `flake8` in CI.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,18 +1,14 @@
 language: python
+dist: xenial
 python:
-- 2.7
-- 3.4
-- 3.5
-- 3.6
-- 3.7-dev
-- pypy
```
Comment (noone234) [inline .travis.yml]: I understand that we're dropping support for Python v2 and older versions 3, but are we also dropping pypy?

File: docs/api.rst [hunk 2]
```diff
@@ -43,30 +43,12 @@ implements the same interface:
          with self.execution_timer('db', 'query', 'foo'):
              rows = yield self.session.query('SELECT * FROM foo')
 
-   .. method:: set_metric_tag(tag, value)
```
Comment (noone234) [inline docs/api.rst:46]: Excellent!

File: examples/statsd.py [hunk 3]
```diff
@@ -14,19 +15,8 @@ class SimpleHandler(statsd.StatsdMixin, web.RequestHandler):
...
     """
 
-    @gen.coroutine
-    def prepare(self):
-        maybe_future = super(SimpleHandler, self).prepare()
-        if concurrent.is_future(maybe_future):
-            yield maybe_future
-
-        if 'Correlation-ID' in self.request.headers:
-            self.set_metric_tag('correlation_id',
```
Comment (noone234) [inline examples/statsd.py:24]: Excellent!
With set_metric_tag() gone, prepare() is no longer necessary. Thanks for simplifying.

File: .travis.yml (hunk 1 above)
Comment (noone234) [inline .travis.yml:8]: I understand that we're dropping support for Python v2 and older versions 3, but are we also dropping pypy?

**Review by noone234 — APPROVED** [review]
We talked.

File: .travis.yml (hunk 1 above)
Comment (andrewrabert) [reply .travis.yml:8]: Yes.

**Pull request merged**